projects/com.nvidia.grcuda/src/com/nvidia/grcuda/gpu/CUDARuntime.java

```diff
@@ -305,17 +303,15 @@ private Object getSymbol(CUDARuntimeFunction function) throws UnknownIdentifierE
                         function.getFunctionFactory().getNFISignature());
     }
 
-    private void checkCUDAReturnCode(Object result, String function) {
+    private void checkCUDAReturnCode(Object result, String... function) {
         if (!(result instanceof Integer)) {
             CompilerDirectives.transferToInterpreter();
-            throw new RuntimeException(
-                            "expected return code as Integer object in " + function + ", got " +
-                                            result.getClass().getName());
+            throw new GrCUDAException("expected return code as Integer object in " + function + ", got " + result.getClass().getName());
```

Comment:
I think there is a problem with `String... function` which turns in to a `String[]` and subsequently into something like `"[Ljava.langString;...."`.